chore(deps): pnpm v11 移行 — onlyBuiltDependencies 設定

Renovate PR #359 で発覚。pnpm 11 から build scripts (postinstall 等) が default で run されなくなった。

## やる事

\`package.json\` に \`pnpm.onlyBuiltDependencies\` を追加 (or \`pnpm approve-builds\` の結果を commit):

\`\`\`json
{
  "pnpm": {
    "onlyBuiltDependencies": [
      "esbuild",
      "sharp",
      "workerd"
    ]
  }
}
\`\`\`

詳細: https://pnpm.io/9.x/cli/approve-builds

POC のうちは pnpm 10 で固定。

**Prep 完了**: PR #367 で package.json に \`pnpm.onlyBuiltDependencies: ["esbuild", "sharp", "workerd"]\` を追加済。これで pnpm v11 再 retry 時の \`[ERR_PNPM_IGNORED_BUILDS]\` は解消される見込み。

Renovate 次の sweep で再 PR 化される時に CI green になるはず。green になったら本 issue は close 可。

--- a/package.json
+++ b/package.json
@@ -2,7 +2,7 @@
   "name": "webull-trading",
   "private": true,
   "type": "module",
-  "packageManager": "pnpm@10.33.4",
+  "packageManager": "pnpm@11.3.0",
   "scripts": {
     "dev": "wrangler dev",
     "deploy": "wrangler deploy",
@@ -28,12 +28,5 @@
     "vite": "^8.0.14",
     "vitest": "^4.0.0",
     "wrangler": "^4.0.0"
-  },
-  "pnpm": {
-    "onlyBuiltDependencies": [
-      "esbuild",
-      "sharp",
-      "workerd"
-    ]
   }
 }

--- a/pnpm-workspace.yaml
+++ b/pnpm-workspace.yaml
@@ -1,0 +1,13 @@
+# pnpm 11+ moved settings out of package.json's "pnpm" block.
+# esbuild / sharp / workerd ship native postinstall scripts that
+# wrangler / vitest / workers require — allow them explicitly.
+allowBuilds:
+  esbuild: true
+  sharp: true
+  workerd: true
+
+# pnpm 11 defaults minimumReleaseAge to 24h as a supply-chain check. That
+# blocks every fresh Renovate lockfile-bump PR for the first day, which this
+# repo can't tolerate during POC. 0 = disabled; revisit once we have a
+# non-bot release gating layer.
+minimumReleaseAge: 0

